cmd/pg/backup_fetch.go

```diff
@@ -40,16 +51,43 @@ var backupFetchCmd = &cobra.Command{
 			pgFetcher = internal.GetPgFetcherOld(args[0], fileMask, restoreSpec)
 		}
 
-		internal.HandleBackupFetch(folder, args[1], pgFetcher)
+		internal.HandleBackupFetch(folder, targetBackupSelector, pgFetcher)
 	},
 }
 
+// create the BackupSelector to select the backup to fetch
+func createTargetBackupSelector(cmd *cobra.Command, args []string) (internal.BackupSelector, error) {
+	switch {
+	case len(args) == 2 && targetUserData != "":
```

Comment:
There is no such a variable or argument as "targetUserData" in this function.

Reply:
It is a global variable. It needs to be global to make it work with cobra command flag declaration.